### PR TITLE
[tensorexpr] Add missing python bindings for NNC Stmts

### PR DESCRIPTION
**Pull request by huiguoo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55570 [tensorexpr] Add missing python bindings for NNC Stmts**

Differential Revision: [D27634987](https://our.internmc.facebook.com/intern/diff/D27634987)

### CODE REVIEW

File: test/test_tensorexpr_pybind.py [hunk 1]
```diff
@@ -40,6 +41,109 @@ def compute(i):
...
+            stmt = torch._C._te.simplify(loopnest.root_stmt())
+            codegen = torch._C._te.construct_codegen('llvm', stmt, [torch._C._te.BufferArg(x) for x in [Pimage, Pweight, Pout]])
+
+            image = torch.ones(N, C, H, W)
+            weight = torch.ones(K, C, R, S)
+            ref= F.conv2d(image, weight)
+            out = torch.zeros_like(ref)
+            codegen.call([image.float(), weight.float(), out.float()])
+            torch.testing.assert_allclose(out, ref)
+
```
Comment (huiguoo) [inline test/test_tensorexpr_pybind.py:146]: Not sure if it's proper to add the tests here. We already have a conv test written in cpp. @ZolotukhinM

**Review by ZolotukhinM — COMMENTED** [review]
Overall looks like a good addition, but please address the linter errors first - it's hard to review the PR appropriately now, the linter is too loud :)

**Comment by huiguoo** [comment]
@ZolotukhinM Removed the conv tests - they are bulky and we have already have one conv test in cpp. btw, is there a way to put this back for review? So hard to find the key.

**Review by ZolotukhinM — APPROVED** [review]
That looks good, thanks! But you will need to rebase to a more recent master: I recently removed masks from Loads and Stores, so you will need to update your PR accordingly before landing it.

> btw, is there a way to put this back for review? So hard to find the key.

There is a "re-request review" button in the reviewers list for each person who has reviewed before.

File: torch/csrc/jit/tensorexpr/tensorexpr_init.cpp [hunk 2]
```diff
@@ -127,7 +127,26 @@ void initTensorExprBindings(PyObject* module) {
...
+      .def(
+          "storeWithMask",
+          [](Placeholder& self,
+             const std::vector<ExprHandle>& args,
+             const ExprHandle& val,
+             const ExprHandle& mask) {
+            return self.storeWithMask(args, val, mask);
+          })
+      .def("buf", [](Placeholder& self) { return BufHandle(self.data()); })
+      .def("data", [](Placeholder& self) { return self.data(); });
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/tensorexpr_init.cpp:149]: I don't think we ever return raw pointers in python?

File: torch/csrc/jit/tensorexpr/tensorexpr_init.cpp [hunk 3]
```diff
@@ -232,12 +275,57 @@ void initTensorExprBindings(PyObject* module) {
...
-      .def("body", &For::body, py::return_value_policy::reference);
+      .def("body", &For::body, py::return_value_policy::reference)
+      .def("set_parallel", &For::set_parallel)
+      .def_static(
+          "make",
+          [](const VarHandle& var,
+             const ExprHandle& start,
+             const ExprHandle& stop,
+             Stmt* body,
+             const LoopOptions& loop_options) {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/tensorexpr_init.cpp:310]: Have we bound `LoopOptions` to python?

**Review by ZolotukhinM — APPROVED** [review]
